packages/tools/devtools/devtools-core/src/data-visualization/VisualSharedTreeTypes.ts

```diff
@@ -27,13 +27,21 @@ export interface SharedTreeSchemaNode {
 	/**
 	 * Name of the SharedTree schema.
 	 */
-	schemaName: string;
+	readonly schemaName: string;
 
 	/**
 	 * Types allowed (e.g., string, number, boolean, handle & etc.) inside the node.
 	 * - InternalNode: `Record<string, string>`.
 	 */
-	allowedTypes?: string | Record<string, string>;
+	readonly allowedTypes?: string | Record<string, string>;
+
+	/**
+	 * If the field is required or optional.
+	 * - When {@link FieldKind.Required}: The field must be present
+	 * - When {@link FieldKind.Optional}: The field may be omitted
+	 * - When undefined: Treated the same as {@link FieldKind.Optional}
+	 */
+	readonly isRequired?: string;
```

Comment:
Is this ever undefined? I think we should always know this, right? Making `undefined` === `optional` seems suspect to me.

Reply:
Made `isRequired` optional since rendering always rendering `true` for the array and map nodes may mislead the users to think the elements in array and map can be optional. 

I can see removing optional type in `allowedTypes` is the right choice since every nodes should render this.

Reply:
Oh, that's right. Sorry, I forgot about our discussion on that 😅 Okay, let's be sure to correct the documentation on this then. Optional means that we shouldn't _display_ the `isRequired` information.

It probably makes sense to make `allowedTypes` a required property until we have a reason to make it optional. Makes usage a bit less error-prone.

Reply:
> always rendering `true` for the array and map nodes may mislead the users to think the elements in array and map can be optional.

Not sure I follow this. Did you mean that it might mislead the users into thinking the elements _can't_ be optional? If so, I'm not sure I agree... like Josh, I lean towards always rendering this. Especially when trying to understand the structure of the tree, I think "implicitness" when displaying things might be the more confusing option. Maybe a trade-off of confusing regarding _what_, but IMO making this field non=optional makes more sense.

Reply:
For both arrays and maps, child data is _always_ required (not optional). Ji and I had been chatting about the visual representation here, and one of us had the thought (I forget who at this point) that always displaying `required: true` might give the false impression that it would be possible for them to be optional. So we figured it might be better to just omit the requirement information in that case.

Curious what your take is, though. We _could_ just always display `required: true` for those cases.

Reply:
Ok I see now. So it's somewhat related to [this other comment](https://github.com/microsoft/FluidFramework/pull/23573#discussion_r1951387189) I made. I guess I'd base my answer on whether we're displaying elements of arrays/maps in a way that's visually similar to the fields of an object node or not; a way which makes it clear that there _is_ a SharedTree field there. If we are, then I think I lean towards including `isRequired: true`. If we're visually displaying them in a more... "friendly" way, where they feel more like an inherent part of the array/map and not separated by a field, I think not displaying `isRequired` does make sense.

Reply:
Yeah. As you've hinted at above, we're using the more "friendly" representation here, which matches the public API surface. In that world, fields are mostly hidden in the context of arrays / maps.